build.sbt

```diff
@@ -20,19 +20,19 @@ resolvers ++= Seq("jitpack" at "https://jitpack.io")
 
 libraryDependencies ++= Seq(
   "org.slf4j" % "slf4j-api" % "1.7.30" % "provided",
-  "com.monitorjbl" % "xlsx-streamer" % "2.1.0"
+  "org.apache.poi" % "poi" % "4.1.0"
 ).map(_.excludeAll(ExclusionRule(organization = "stax")))
 
 shadedDeps ++= Seq(
-  "org.apache.poi" ^ "poi" ^ "4.1.0",
```

Comment:
Are you sure we don't need to shade POI as well?

Reply:
Not sure, I'm not an expert of shading. But it worked.
The dependency graph looks like below:
poi (4.1.0)
xlsx-streamer -> poi (4.0.0)

So when only poi was shaded, xlsx-streamer was searching for shaded poi, but couldn't find it.
We can try to shade both if needed.

Reply:
Right now I would say let's shade both. Afaik, the only downside is that the size of the JAR increases. Other than that it should help if someone has a (possibly transitive) dependency on POI with a different version.